test/tests.js

```diff
@@ -29127,3 +29127,141 @@ test("/[a-z]/gim", {
 testFail("/[a-z]/u", "Invalid regular expression flag (1:1)");
 testFail("/[a-z]/y", "Invalid regular expression flag (1:1)");
 testFail("/[a-z]/s", "Invalid regular expression flag (1:1)");
+
+// https://github.com/ternjs/acorn/issues/255
+
+test("(a) = {}", {
```

Comment:
> and it might be a case for going through checkPatternErrors after all)

Exactly, such patterns were a reason why this particular issue was not as simple to fix as it seemed.

Although I'm happy to see progress on it, thanks @kaicataldo for working on this!